### ISSUE TITLE
Reverb not working correctly

### ISSUE THREAD

**Issue by Ripper69** [PCSX2/pcsx2#2093]
***PCSX2 version:***
1.5.0-dev-2205-gaf-2278c3c

***PCSX2 options:***
SPU2-X  catmul rom, didnt tick any boxes, xaudio 2.7, volume 100%, latency 100 ms, syncronizing mode timestretch, audio expansion mode stereo

***Plugins used:***

SPU2-X

***Description of the issue:***
Reverb and maybe other audio effects don't play correctly in Final Fantasy X's music

Comparison:

PCSX2: https://soundcloud.com/josh-bales-209156607/2a-1

PS2: https://soundcloud.com/josh-bales-209156607/3a-1

***PC specifications:***
I7 4770 3.4 ghz, GTX 770

**Comment by MrCK1** [comment]
I have good ears, I'll take a listen later.

**Comment by Ripper69** [comment]
On soundcloud you could click on more and select download, the original file is FLAC

**Comment by willkuer** [comment]
I think @sudonim1 was the last one working on it. But I think he gave up for the moment after having a larger improvement of the reverb emulation.

**Comment by Ripper69** [comment]
Here is the track without reverb: https://soundcloud.com/josh-bales-209156607/4a-1

**Comment by MrCK1** [comment]
After listening to the new track, it turns out that the reverb is still present as it should be in PCSX2. I guess the problem lies now with the intensity of it. Whether that's something we are able to address is unknown.

**Comment by ramapcsx2** [comment]
Well, it's a bit hard to tell but I think our reverb is cutting into the sample with early reflections.
So yea, this helps reduce the factors a little.
http://problemkaputt.de/psx-spx.htm#spureverbformula

**Comment by Mrlinkwii** [comment]
is this still an issue on latest master?

**Comment by Mrlinkwii** [comment]
still an issue on the nightly's

**Comment by seta-san** [comment]
> After listening to the new track, the reverb is definitely still present as it should be in PCSX2. The problem lies now with the intensity of attack as @ramapcsx2 mentioned. Whether that's something we are able to address is unknown.

i'm notoriously bad at telling the difference between audio. between the rework on the reverb logic and ADSR logic does this sound better? https://github.com/PCSX2/pcsx2/pull/10084

**Comment by bigol83** [comment]
This is how it sound like in latest nightly 1.7.5120 with default settings. 


https://github.com/PCSX2/pcsx2/assets/38129260/0c18c221-2e76-4d7d-937e-d2e36a1e1ca9

**Comment by seta-san** [comment]
would you say it's fixed or just better?

**Comment by bigol83** [comment]
It's better in reverb, but it seems like bass is less noticeable, compared to both Pcsx2 1.5 sample and Ps2 sample.

**Comment by Mrlinkwii** [comment]
can this be checked on latest nightly

**Comment by bigol83** [comment]
This is how it sounds in Pcsx2 1.7.5259

https://github.com/PCSX2/pcsx2/assets/38129260/d6b15fbc-69df-43e1-95f7-7fd421730954

**Comment by bigol83** [comment]
The graph shows that 5120 and 5259 are exactly the same. Bass is definitely less noticeable compared to PS2 and early Pcsx2 versions.

**Comment by Jobima1st** [comment]
yes I note that in mgs games too, when you go to game menu https://youtu.be/srb_8SPx47c?t=216 (not only in this video but in any one from real ps2 like https://youtu.be/DHSH8O4QhlE?t=2), in pcsx2 the audio is not like this!

this was the same problem in ps1 emulations https://www.youtube.com/watch?v=6WQwU6zHUOQ but Mednafen fix it https://www.youtube.com/watch?v=lTNZ9GwhgQU then others take the fix too

**Comment by Jobima1st** [comment]
I just did some test with [Play!](https://purei.org/downloads.php) and the Reverb in mgs3 is better and closer to the real ps2, maybe pcsx2 can idea from Play!

**Comment by Ziemas** [comment]
Looks like Play uses the same old algorithm basically every old PS1 emulator used, I'm pretty sure people think that one is worse than what Mednafen has (which is what PCSX2's is based on).

**Comment by Jobima1st** [comment]
wasn't every old PS1 emulator Reverb was almost not work? so in this case I think it's opposite between Play! and PCSX2 since even last nightly PCSX2 almost got zero Reverb in mgs3 while Play! sound is not as good in quality but it seems do the Reverb right at least in mgs3
